### PR TITLE
Dit 2695 - delete user journey

### PR DESCRIPTION
**Pull request by oliwiaKot**
Do not merge before DIT-2691
Created views for a confirmation page and the "cases must be moved" page and a controller endpoint to post the confirmation to and delete the user.

### CODE REVIEW

File: app/service/UserService.scala [hunk 1]
```diff
@@ -43,5 +43,12 @@ class UserService @Inject() (
...
       userDetails <- connector.getUserDetails(id)
-    } yield userDetails
+    } yield {
+      if (userDetails.deleted) {
+        throw new Exception("User with pid " + id + " is deleted")
+      } else {
+        userDetails
+      }
+    }
+
```
Comment (DavidGregory084) [inline app/service/UserService.scala:53]: Following up from my comment on the other PR, I think this should return `Option` and filter out `deleted` users as `None`, i.e.

```scala
connector
  .getUserDetails(id) // The connector method returns Future[Option[Operator]]
  .map(_.filterNot(_.deleted)) // Filter out deleted users in the inner Option
```

File: app/controllers/v2/ManageUserController.scala [hunk 2]
```diff
@@ -56,4 +60,47 @@ class ManageUserController @Inject() (
...
+            for {
+              user <- userService.getUser(pid)
+            } yield Ok(confirmDeleteUser(user, errors)), {
+            case true =>
+              for {
+                user <- userService.getUser(pid)
+              } yield {
+                val updatedUser = user.copy(deleted = true)
+                userService
+                  .updateUser(updatedUser, request.operator)
```
Comment (DavidGregory084) [inline app/controllers/v2/ManageUserController.scala:95]: Did we add a markDeleted endpoint in the backend? Ignore me if not

**Review by DavidGregory084 — APPROVED** [review]
Looks great 👍  just got some more suggestions relating to the `getUserDetails` API

File: app/service/UserService.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/service/UserService.scala:47]: yeah you're right, done this - thanks a lot!

File: app/controllers/v2/ManageUserController.scala (hunk 2 above)
Comment (oliwiaKot) [reply app/controllers/v2/ManageUserController.scala]: We did, my bad I didn't notice this, I added a service and connector method to use markDeleted and call that in the controller now

File: app/service/UserService.scala [hunk 3]
```diff
@@ -41,5 +41,15 @@ class UserService @Inject() (
...
+    for {
+      userDetails <- connector.getUserDetails(id).map(_.filterNot(_.deleted))
+    } yield userDetails
+
+  def markDeleted(user: Operator, operatorMakingTheChange: Operator)(
+    implicit hc: HeaderCarrier
+  ): Future[Operator] =
+    for {
+      deleted <- connector.markDeleted(user)
+      _ = auditService.auditUserUpdated(user, operatorMakingTheChange)
```
Comment (DavidGregory084) [inline app/service/UserService.scala:53]: I think this should provide `deleted` to the audit function

Comment (DavidGregory084) [reply app/service/UserService.scala:53]: Ah I see we only use the `id` of the operator in the audit payload so this probably doesn't matter

**Review by DavidGregory084 — APPROVED** [review]
👍

File: app/service/UserService.scala (hunk 3 above)
Comment (oliwiaKot) [reply app/service/UserService.scala]: yeah it uses the id, but I changed that anyway, good spot!

**Review by vajarova — APPROVED** [review]

**Pull request merged**